src/main/java/io/gravitee/policy/transformheaders/TransformHeadersPolicy.java

```diff
@@ -76,6 +87,63 @@ public void onResponse(Request request, Response response, ExecutionContext exec
         policyChain.doNext(request, response);
     }
 
+    @OnRequestContent
+    public ReadWriteStream<Buffer> onRequestContent(ExecutionContext executionContext) {
+        if (transformHeadersPolicyConfiguration.getScope() == PolicyScope.REQUEST_CONTENT) {
+            return createStream(PolicyScope.REQUEST_CONTENT, executionContext);
+        }
+
+        return null;
+    }
+
+    @OnResponseContent
+    public ReadWriteStream<Buffer> onResponseContent(ExecutionContext executionContext) {
+        if (transformHeadersPolicyConfiguration.getScope() == PolicyScope.RESPONSE_CONTENT) {
+            return createStream(PolicyScope.RESPONSE_CONTENT, executionContext);
+        }
+
+        return null;
+    }
+
+    private ReadWriteStream<Buffer> createStream(PolicyScope scope, ExecutionContext context) {
+        return new BufferedReadWriteStream() {
+            io.gravitee.gateway.api.buffer.Buffer buffer = io.gravitee.gateway.api.buffer.Buffer.buffer();
+
+            @Override
+            public SimpleReadWriteStream<Buffer> write(io.gravitee.gateway.api.buffer.Buffer content) {
+                buffer.appendBuffer(content);
+                return this;
+            }
+
+            @Override
+            public void end() {
+                initRequestResponseProperties(
+                    context,
+                    (scope == PolicyScope.REQUEST_CONTENT) ? buffer.toString() : null,
+                    (scope == PolicyScope.RESPONSE_CONTENT) ? buffer.toString() : null
+                );
+
+                if (scope == PolicyScope.REQUEST_CONTENT) {
+                    transform(context.request().headers(), context);
+                } else {
+                    transform(context.response().headers(), context);
+                }
```

Comment:
It ends in timeout without that (both with postman or debug mode):

```suggestion
                }
                
                if (buffer.length() > 0) {
                    super.write(buffer);
                }

                super.end();
```